### ISSUE TITLE
highlight.js vulnerable to Prototype Pollution

### ISSUE THREAD

**Issue by shedd** [RenoFi/react-tooltip#238]
GHSA-vfrc-7r7c-w9mx
low severity
Vulnerable versions: < 9.18.2
Patched version: 9.18.2

Remediation
Upgrade highlight.js to version 9.18.2 or later. For example:
```
highlight.js@^9.18.2:
  version "9.18.2"
```

Impact
Affected versions of this package are vulnerable to Prototype Pollution. A malicious HTML code block can be crafted that will result in prototype pollution of the base object's prototype during highlighting. If you allow users to insert custom HTML code blocks into your page/app via parsing Markdown code blocks (or similar) and do not filter the language names the user can provide you may be vulnerable.

The pollution should just be harmless data but this can cause problems for applications not expecting these properties to exist and can result in strange behavior or application crashes, i.e. a potential DOS vector.

If your website or application does not render user provided data it should be unaffected.

Patches
Versions 9.18.2 and 10.1.2 and newer include fixes for this vulnerability. If you are using version 7 or 8 you are encouraged to upgrade to a newer release.

Workarounds
Patch your library
Manually patch your library to create null objects for both languages and aliases:
```
const HLJS = function(hljs) {
  // ...
  var languages = Object.create(null);
  var aliases = Object.create(null);
```

Filter out bad data from end users
Filter the language names that users are allowed to inject into your HTML to guarantee they are valid.

References
What is Prototype Pollution?
highlightjs/highlight.js#2636
For more information
If you have any questions or comments about this advisory:

Please file an issue against highlight.js

**Comment by shedd** [comment]
Dependabot cannot update highlight.js to a non-vulnerable version
The latest possible version that can be installed is 9.15.10 because of the following conflicting dependencies:

```
@storybook/addon-storysource@6.0.28 requires highlight.js@~9.15.1 via a transitive dependency on react-syntax-highlighter@12.2.1
@storybook/addon-storysource@6.0.28 requires highlight.js@~9.15.0 via a transitive dependency on lowlight@1.12.1
@storybook/react@6.0.28 requires highlight.js@~9.15.1 via a transitive dependency on react-syntax-highlighter@12.2.1
@storybook/react@6.0.28 requires highlight.js@~9.15.0 via a transitive dependency on lowlight@1.12.1
```

The earliest fixed version is 10.1.2.

**Comment by shedd** [comment]
Resolved in https://github.com/RenoFi/react-tooltip/pull/239
